### DISCUSSION TITLE
Why is the throwError operator depreciated?

### DISCUSSION THREAD

**Discussion by mlc-mlapis** [ReactiveX/rxjs#7312] [Q&A] [answered]
The operator `throwError` allows you to return any object you want. When its depreciation was announced, the recommended alternative was `new Error('.....'),` but it means that only a string can be returned. Am I the only one who needs to return a more structured error, like `{errorCode: ..., errorMessage: ..., metadata: ...}`?

**Comment by demensky** [comment, marked as answer]
Only the overload where the error object is passed directly has been deprecated.

```diff
-throwError(new Error())
+throwError(() => new Error())
   .subscribe({
     error: (error) => {
       console.log(error);
     }
   });
```

**Reply by mlc-mlapis** [reply]
Ops, thanks. It hasn't been clear from the first moment.
